workos/session.py

```diff
@@ -0,0 +1,174 @@
+        self.jwk_algorithms = ["RS256"]
+
+    def authenticate(
+        self,
+    ) -> Union[
+        AuthenticateWithSessionCookieSuccessResponse,
+        AuthenticateWithSessionCookieErrorResponse,
+    ]:
+        if self.session_data is None:
+            return AuthenticateWithSessionCookieErrorResponse(
+                authenticated=False,
+                reason=AuthenticateWithSessionCookieFailureReason.NO_SESSION_COOKIE_PROVIDED,
+            )
+
+        try:
+            session = self.unseal_data(self.session_data, self.cookie_password)
+        except Exception:
+            return AuthenticateWithSessionCookieErrorResponse(
+                authenticated=False,
+                reason=AuthenticateWithSessionCookieFailureReason.INVALID_SESSION_COOKIE,
+            )
+
+        if not session["access_token"]:
+            return AuthenticateWithSessionCookieErrorResponse(
+                authenticated=False,
+                reason=AuthenticateWithSessionCookieFailureReason.INVALID_SESSION_COOKIE,
+            )
+
+        if not self.is_valid_jwt(session["access_token"]):
+            return AuthenticateWithSessionCookieErrorResponse(
+                authenticated=False,
+                reason=AuthenticateWithSessionCookieFailureReason.INVALID_JWT,
+            )
+
+        signing_key = self.jwks.get_signing_key_from_jwt(session["access_token"])
+        decoded = jwt.decode(
+            session["access_token"], signing_key.key, algorithms=self.jwk_algorithms
+        )
+
+        return AuthenticateWithSessionCookieSuccessResponse(
+            authenticated=True,
+            session_id=decoded["sid"],
+            organization_id=decoded.get("org_id", None),
+            role=decoded.get("role", None),
+            permissions=decoded.get("permissions", None),
+            entitlements=decoded.get("entitlements", None),
+            user=session["user"],
+            impersonator=session.get("impersonator", None),
+        )
+
+    def refresh(self, options: Optional[Dict[str, Any]] = None) -> Union[
+        RefreshWithSessionCookieSuccessResponse,
+        RefreshWithSessionCookieErrorResponse,
+    ]:
+        cookie_password = (
+            self.cookie_password if options is None else options.get("cookie_password")
+        )
+        organization_id = None if options is None else options.get("organization_id")
+
+        try:
+            session = self.unseal_data(self.session_data, cookie_password)
+        except Exception:
+            return RefreshWithSessionCookieErrorResponse(
+                authenticated=False,
+                reason=AuthenticateWithSessionCookieFailureReason.INVALID_SESSION_COOKIE,
+            )
+
+        if not session["refresh_token"] or not session["user"]:
+            return RefreshWithSessionCookieErrorResponse(
+                authenticated=False,
+                reason=AuthenticateWithSessionCookieFailureReason.INVALID_SESSION_COOKIE,
+            )
```

Comment:
Same here regarding `refresh_token`/`user`, should we do `.get(<key>, None)` for safety?